apluslms_roman/cli.py

```diff
@@ -379,6 +413,142 @@ def config_print_action(context):
+                        val = int(val)
+                    else:
+                        raise TypeError("'" + key + "' should be an integer.")
+                elif val_type == "boolean":
+                    if val.lower == "true" or val == "1":
+                        val = True
+                    elif val.lower == "false" or val == "0":
+                        val = False
+                    else:
+                        raise TypeError("'" + key + "' should be a boolean.")
+            except KeyError:
+                raise KeyError("'" + key + "' is not a key found in the schema.")
+            document.mlset(key, val)
+        except ValueError:
+            raise ValueError("Give values in format 'key=val'.")
+    document.validate()
+    document.save()
+
+def del_args(context):
+    if context.args.user == context.args.project:
+        raise Exception("Choose either user settings or project settings")
+    if context.args.user:
+        document = context.settings
+    else:
+        document = context.config
+    for val in context.args.keys:
+        document.mldel(val)
+    document.validate()
+    document.save()
+
+def add_step(context):
+    args = context.args
+    env = args.env
+    if env:
+        try:
+            env = {a: b for a, b in [s.split('=') for s in env]}
+        except ValueError:
+            raise ValueError("env is a dict, so values need to be in key=val format, e.g. a=1 b=2")
+    step = {
+        "img": args.img,
+        "cmd": args.cmd,
+        "mnt": args.mnt,
+        "env": env,
+        "name": args.name
+    }
+    nulls = [k for k in step if not step[k]]
+    for k in nulls:
+        del step[k]
+    for k in step:
+        if k != "env":
+            step[k] = step[k][0]
+    config = get_config(context)
+    if "name" in step:
+        other_names = [s["name"] for s in [s.get_data() for s in config.steps] if "name" in s]
+        if step["name"] in other_names:
+            raise ValueError("A step with the name '" + step["name"] + "' already exists.")
+    config.steps.append(step)
+    config.validate()
+    config.save()
+    print("Step successfully added to config.")
+
+def del_step(context):
+    name = context.args.ref[0].lower()
+    config = get_config(context)
+    steps = config.steps
+    if name.isdigit():
+        idx = int(name)
+        if not confirm_del(steps[idx]):
+            return
+        del steps[idx]
+    else:
+        names = [s["name"].lower() for s in [s.get_data() for s in config.steps] if "name" in s]
+        if name not in names:
+            raise ValueError("There is no step named '" + name + "'.")
+        step = [
+            s for s in config.steps
+            if "name" in s.get_data() and s.get_data()["name"] == name
+        ][0]
+        if not confirm_del(step):
+            return
+        config.steps.remove(step)
+    config.validate()
+    config.save()
+    print("Step successfully removed from config.")
+
+def confirm_del(step):
```

Comment:
this could be inside `del_step` and it should be possible to skip it with `-f` / `--force` flag.